jobs/kpi-forecasting/kpi_forecasting.py

```diff
@@ -11,22 +25,160 @@
 }
 
 
+class KPIPipeline:
+    def __init__(self, config_path):
+        self.config_data = load_yaml(filepath=config_path)
+        model_type = self.config_data["model_type"]
+        if model_type == "funnel":
+            self.model_type = "funnel"
+            self.model_class = FunnelForecast
+            self.segments = list(self.config_data["metric_hub"]["segments"].keys())
+        elif model_type == "prophet":
+            self.model_type = "prophet"
+            self.model_class = ProphetForecast
+            self.segments = None
+        else:
+            raise ValueError(f"Model type {model_type} is not supported")
+
+    def add_metadata(self, summary_df):
+        # add Metric Hub metadata columns
+        summary_df["metric_alias"] = self.metric_hub.alias.lower()
+        summary_df["metric_hub_app_name"] = self.metric_hub.app_name.lower()
+        summary_df["metric_hub_slug"] = self.metric_hub.slug.lower()
+        summary_df["metric_start_date"] = pd.to_datetime(self.metric_hub.min_date)
+        summary_df["metric_end_date"] = pd.to_datetime(self.metric_hub.max_date)
+        summary_df["metric_collected_at"] = self.collected_at
+
+        # add forecast model metadata columns
+        summary_df["forecast_start_date"] = self.start_date
+        summary_df["forecast_end_date"] = self.end_date
+        summary_df["forecast_trained_at"] = self.trained_at
+        summary_df["forecast_predicted_at"] = self.predicted_at
+
+        return summary_df
+
+    def get_raw_data(self):
```

Comment:
I think "raw" may be too ambiguous here.
```suggestion
    def get_historical_data(self):
```